manager/integration/tests/test_node.py

```diff
@@ -0,0 +1,375 @@
+    return disk_path
+
+
+def clean_disk(client, *args):
+    # clean disk
+    for vol_name in args:
+        # umount disk
+        common.umount_host_disk(vol_name)
+        # clean volume
+        clean_volume(client, vol_name)
+
+    volumes = client.list_volume()
+    assert len(volumes) == 0
+    # clean all disks
+    common.cleanup_host_disk_directory()
+
+
+def get_update_disks(disks):
+    update_disk = []
+    for key, disk in disks.iteritems():
+        update_disk.append(disk)
+    return update_disk
+
+
+def test_node(clients):  # NOQA
+    # get a random client
+    for host_id, client in clients.iteritems():
+        break
+
+    # test node update
+    nodes = client.list_node()
+    assert len(nodes) > 0
+    lht_hostId = get_self_host_id()
+    node = client.by_id_node(lht_hostId)
+    node = client.update(node, allowScheduling=False)
+    assert not node["allowScheduling"]
+    node = client.by_id_node(lht_hostId)
+    assert not node["allowScheduling"]
+
+    node = client.update(node, allowScheduling=True)
+    assert node["allowScheduling"]
+    node = client.by_id_node(lht_hostId)
+    assert node["allowScheduling"]
+
+    disks = node["disks"]
+
+    # test add disk exception
+    with pytest.raises(Exception):
+        # add same disk by different mount path
+        disk = {"path": "/var/lib", "allowScheduling": True,
+                "storageMaximum": 5*Gi, "storageReserved": 2*Gi}
+        update_disk = get_update_disks(disks)
+        update_disk.append(disk)
+        node = node.diskUpdate(disks=update_disk)
+
+    # test delete disk exception
+    with pytest.raises(Exception):
+        node.diskUpdate(disks=[])
+
+    # create multiple disks for node
+    node = client.by_id_node(lht_hostId)
+    disks = node["disks"]
+    disk_path1 = create_disk(client, 'vol-disk-1',
+                             SIZE, lht_hostId)
+    disk1 = {"path": disk_path1, "allowScheduling": True,
+             "storageMaximum": 5*Gi, "storageReserved": 2*Gi}
+    disk_path2 = create_disk(client, 'vol-disk-2',
+                             SIZE, lht_hostId)
+    disk2 = {"path": disk_path2, "allowScheduling": True}
+
+    update_disk = get_update_disks(disks)
+    # add new disk for node
+    update_disk.append(disk1)
+    update_disk.append(disk2)
+
+    # save disks to node
+    node = node.diskUpdate(disks=update_disk)
+    assert len(node["disks"]) == len(update_disk)
+
+    node = client.by_id_node(lht_hostId)
+    assert len(node["disks"]) == len(update_disk)
+
+    # update disk
+    disks = node["disks"]
+    update_disk = get_update_disks(disks)
+    for disk in update_disk:
+        # keep default disk for other tests
+        if disk["path"] == disk_path1 or disk["path"] == disk_path2:
+            disk["allowScheduling"] = False
+            disk["storageReserved"] = Gi
+    node.diskUpdate(disks=update_disk)
+    # wait for node controller to update disk status
+    time.sleep(WAIT_FOR_NODE_STATUS)
```

Comment:
Can we use polling (wait for the expected result) instead of sleep here (and below)? Interval can be longer but wait for 1 minutes everytime seems excessive.